internal/appsec/listener/sharedsec/shared.go

```diff
@@ -82,6 +82,18 @@ func AddWAFMonitoringTags(th tagsHolder, rulesVersion string, overallRuntimeNs,
 	th.AddTag(wafDurationExtTag, float64(overallRuntimeNs)/1e3) // ns to us
 }
 
+// AddAPISecurityTags serializes the WAF derivatives and adds them to the tags
+func AddAPISecurityTags(th tagsHolder, derivatives map[string]any) {
+	for k, v := range derivatives {
+		schema, err := json.Marshal(v)
+		if err != nil {
+			log.Debug("appsec: could not serialize API Security schema for %s: %v", k, err)
+			continue
+		}
+		th.AddTag(k, string(schema))
+	}
```

Comment:
I think we must keep minimizing these expensive json calls and doing it once at the end of the request like we do for the events. Can you please save the derivatives in memory in the request operation and marshal them once at the end of the request?

This helps lowering the memory pressure, saving CPU time in the request path, and not impacting the response time because we do it post-response.

Reply:
I added the ability for tag holders to hold serializable tags that get marshalled when calling `Tags()`. Seems like a pretty smooth and practical way to do it. 8371afa313c2790ed